applications/admin.py

```diff
@@ -0,0 +1,201 @@
+    search_fields = ("user__email", "user__username")
+    list_filter = ("bootcamp_run__bootcamp",)
+    raw_id_fields = ("user", "bootcamp_run", "order")
+
+    def get_queryset(self, request):
+        """Overrides base queryset"""
+        return (
+            super().get_queryset(request).select_related("user", "bootcamp_run__bootcamp")
+        )
+
+    def get_user_email(self, obj):
+        """Returns the user email"""
+        return obj.user.email
+
+    get_user_email.short_description = "User"
+    get_user_email.admin_order_field = "user__email"
+
+    def get_bootcamp_title(self, obj):
+        """Returns the bootcamp title"""
+        return obj.bootcamp_run.bootcamp.title
+
+    get_bootcamp_title.short_description = "Bootcamp"
+    get_bootcamp_title.admin_order_field = "bootcamp_run__bootcamp__title"
+
+    def get_run_title(self, obj):
+        """Returns the bootcamp run title"""
+        return obj.bootcamp_run.title
+
+    get_run_title.short_description = "Bootcamp Run"
+    get_run_title.admin_order_field = "bootcamp_run__title"
+
+
+class SubmissionTypeAdmin(TimestampedModelAdmin):
+    """Base admin class for submission types"""
+    list_display = ('id', 'get_user_email', 'get_run_title')
+    readonly_fields = ('get_user_email', 'get_run_title')
+
+    def get_queryset(self, request):
+        """Overrides base queryset"""
+        return (
+            super().get_queryset(request).prefetch_related(
+                "app_step_submissions__bootcamp_application__user",
+                "app_step_submissions__bootcamp_application__bootcamp_run",
+            )
+        )
+
+    def get_user_email(self, obj):
+        """Returns the user email"""
+        app_step_submission = obj.app_step_submissions.first()
+        return app_step_submission.bootcamp_application.user.email
+
+    get_user_email.short_description = "Application User"
+    get_user_email.admin_order_field = "app_step_submission__bootcamp_application__user__email"
+
+    def get_run_title(self, obj):
+        """Returns the bootcamp run title"""
+        app_step_submission = obj.app_step_submissions.first()
+        return app_step_submission.bootcamp_application.bootcamp_run.title
+
+    get_run_title.short_description = "Application Bootcamp Run"
+    get_run_title.admin_order_field = "app_step_submission__bootcamp_application__bootcamp_run__title"
+
+
+class VideoInterviewSubmissionAdmin(SubmissionTypeAdmin):
+    """Admin class for VideoInterviewSubmission"""
+    model = models.VideoInterviewSubmission
+
+    def get_list_display(self, request):
+        return tuple(super().get_list_display(request) or ()) + ("video_file",)
+
+
+class QuizSubmissionAdmin(SubmissionTypeAdmin):
+    """Admin for QuizSubmission"""
+    model = models.QuizSubmission
+
+    def get_list_display(self, request):
+        return tuple(super().get_list_display(request) or ()) + ("started_date",)
+
+
+class ApplicationStepSubmissionAdmin(TimestampedModelAdmin):
```

Comment:
Is there a way to add validation that `content_object` exists before updating it? If the user enters an incorrect id number, the page crashes and they can't edit it to correct it.

Reply:
Right, I meant to address this. My feeling is that object id shouldn't be editable at all